File: src/WebUI/ClientApp/src/app/models/components/product.ts
import { Category } from "./category";

export class Product {
    id: number;
    name: string;
    price: number;
    imageUrl: string;
    category: Category;
    isFavorite?: boolean
    
    constructor(){
        this.id = 0;
        this.name  ="";
        this.price = 0;
        this.imageUrl  ="";
        this.name  ="";
        this.category = new Category();
    }
}

File: src/Api/Models/Product.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;
using RiskCenterStoreApi.Enumerations;


namespace RiskCenterStoreApi.Models
{
    [Serializable]
    [Table("products")]
    public class Product
    {
        [Key]
        public int id { get; set; }
        public string name { get; set; }
        public double price { get; set; }
        [ForeignKey("category_id")]
        [Column("category_id")]
        public ProductCategory Category { get; set; }
        [Column("url_image")]
        public string imageUrl { get; set; }
    }
}
